### PR TITLE
feat(VisualTests): Draft/proposal for visual testing

### PR DESCRIPTION
**Pull request by kosanna**
- Added CasperJS (PhantomFlow)
- Added couple of tests on Header to demo the idea (click some links and generate desktop and mobile view reports)

Picks up visual changes if something has been changed, up to a dev wether to rebase screenshot to a newer version or to fix what's been done in a wrong way.

Eg: intentionally changed size of svg here, Casper failed and show the diff:

image

Don't like it - trash it, like it - let's develop it further.

Prerequisites: style-guide should be up and running on 3000.

To run visual tests `npm run casper`.
To see/rebase reports `npm run casper-report`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,25 +4,27 @@
   "description": "Living style guide containing the building blocks and design principles for SEEK web applications",
```
Comment (kosanna) [inline package.json]: Soz for sorting again, lines 9, 10, 110 are new

**Comment by markdalgleish** [comment]
@kosanna I'm having issues getting this running on my local machine. I noticed that these tests aren't running in CI—would you be able to make this run as part of `npm test` to ensure it works from a fresh install?

**Comment by kosanna** [comment]
Yeah, I just was thinking first having it separately, not wired up to `npm t` and then if we go with it, then add it there. It's interesting, did you do `yarn`/`npm i`, may be something didn't install..hmm

**Comment by kosanna** [comment]
Interestingly visual tests pass locally but fail in travis :/

**Comment by markdalgleish** [comment]
@kosanna have you had a chance to look into this further?

**Comment by kosanna** [comment]
@markdalgleish soz, didn't have time for that so far. I will try to add docker with casperjs image to that one as I come back to here.

**Pull request closed without merging**